pkgs/development/python-modules/pyfdt/default.nix

```diff
@@ -0,0 +1,25 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+}:
+
+buildPythonPackage rec {
+  pname = "pyfdt";
+  version = "0.3";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "pyfdt";
+    hash = "sha256:1w7lp421pssfgv901103521qigwb12i6sk68lqjllfgz0lh1qq31";
+  };
+
+  pythonImportsCheck = [ "pyfdt" ];
+
+  meta = {
+    homepage = "https://github.com/superna9999/pyfdt";
+    description = "Flattened device tree parser";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ ralismark ];
+  };
+}
```

Comment:
```suggestion
{
  lib,
  bison,
  buildPythonPackage,
  fetchFromGitHub,
  flex,
  pytestCheckHook,
  setuptools,
}:

buildPythonPackage rec {
  pname = "pyfdt";
  version = "0.3";
  pyproject = true;

  src = fetchFromGitHub {
    owner = "superna9999";
    repo = "pyfdt";
    rev = "refs/tags/pyfdt-${version}";
    hash = "sha256-x5RQI4giQVywCbBSL5MOpa2k0E88gj+NDRMjoja4rQM=";
    fetchSubmodules = true;
  };

  build-system = [ setuptools ];

  nativeCheckInputs = [ flex bison ];

  checkPhase = ''
    runHook preCheck
    bash tests/do_tests.bash
    runHook postCheck
  '';

  # https://github.com/superna9999/pyfdt/issues/21
  # dtc-parser.tab.o:/build/source/tests/dtc/dtc-parser.tab.c:1079: multiple definition of `yylloc'
  doCheck = false;

  pythonImportsCheck = [ "pyfdt" ];

  meta = {
    homepage = "https://github.com/superna9999/pyfdt";
    description = "Flattened device tree parser";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ ralismark ];
  };
}
```

Reply:
thanks for the suggestion! however, given that the those tests don't work, and aren't included in the pypi package, i'll just leave them out. pyfdt hasn't had any code changes in years and it's being used without problems